packages/react/src/components/timeInput/TimeInput.tsx

```diff
@@ -0,0 +1,317 @@
+        event.preventDefault();
+        minutesInputRef.current.focus();
+      }
+      // Increase/decrease the value with arrow up/down keys
+      if (['ArrowUp', 'ArrowDown'].includes(event.key) && !event.shiftKey) {
+        event.preventDefault();
+        const modifier = event.key === 'ArrowUp' ? 1 : -1;
+        const hoursAsInt = parseInt(hours, 10) || 0;
+        const newHours = zeroPad(`${incrementNumber(0, 23, hoursAsInt, modifier)}`);
+        updateTimeInput(newHours, minutes);
+      }
+    };
+
+    /**
+     * Handle keydown event on minutes input
+     */
+    const onMinutesKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
+      // Move to the hours input with left arrow key
+      if (event.key === 'ArrowLeft' && !event.shiftKey) {
+        event.preventDefault();
+        hoursInputRef.current.focus();
+      }
+      // Increase/decrease the value with arrow up/down keys
+      if (['ArrowUp', 'ArrowDown'].includes(event.key) && !event.shiftKey) {
+        event.preventDefault();
+        const modifier = event.key === 'ArrowUp' ? 1 : -1;
+        const minutesAsInt = parseInt(minutes, 10) || 0;
+        const newMinutes = zeroPad(`${incrementNumber(0, 59, minutesAsInt, modifier)}`);
+        updateTimeInput(hours, newMinutes);
+      }
+    };
+
+    /**
+     * Format hours on blur
+     */
+    const onHoursBlur: React.FocusEventHandler = () => {
+      if (hours.length > 0) {
+        updateTimeInput(zeroPad(hours), minutes);
+      }
+    };
+
+    /**
+     * Format minutes on blur
+     */
+    const onMinutesBlur: React.FocusEventHandler = () => {
+      if (minutes.length > 0) {
+        updateTimeInput(hours, zeroPad(minutes));
+      }
+    };
+
+    // Compose aria-describedby attribute
+    const ariaDescribedBy = [helperText && `${id}-helper`, errorText && `${id}-error`, successText && `${id}-success`]
+      .filter((item) => item)
+      .join(' ');
+
+    // Compose props for the input frame
+    const frameProps = {
+      className: classNames(textInputStyles.input, styles.timeInputFrame, disabled && styles.disabled),
+      onClick: (event: React.MouseEvent) => {
+        if (event.target !== hoursInputRef.current && event.target !== minutesInputRef.current) {
+          hoursInputRef.current.focus();
+        }
+      },
+    };
+
+    const hourInputId = `${id}-hours`;
+    const minuteInputId = `${id}-minutes`;
+
+    return (
+      <InputWrapper {...wrapperProps} id={`${id}-hours`}>
+        <div {...frameProps}>
+          <input
+            aria-hidden
+            readOnly
+            className={styles.fullInput}
+            disabled={disabled}
+            id={id}
+            onChange={onChange}
+            ref={inputRef}
+            required={required}
+            type={type}
+            tabIndex={0}
+            value={time}
+            {...rest}
+          />
+          <label htmlFor={hourInputId} className={styles.partialInputLabel}>
+            hours
```

Comment:
Could/should we provide this in different languages? fi, sv and en? We could pass a language prop to the component, and render this label based on it?

Reply:
I added hoursLabel and minutesLabel props.